[CALCITE-6001] Add dialect-specific character sets for encoding string literals

tanclary opened the pull request:
… to allow encoding of non-ISO-8859-1 characters

Going to mark this as a draft for now while I improve implementation and testing.

julianhyde committed:
Workaround

julianhyde committed:
Looker instructions

julianhyde committed:
Temporary workaround: change SqlInternalOperator's syntax back to FUNCTION

Commit:
[CALCITE-5052] Allow Source based on a URL with jar: protocol

This allows dependent projects to run tests using Bazel. (Previously,
DiffRepository would give errors because Bazel has packaged the .xml
files it needs inside JAR files.)

Close apache/calcite#2750

tjbanghart committed:
Use first datetime operand as DATE_TRUNC return type

tjbanghart committed:
[CALCITE-3312] WIP: TIMESTAMPDIFF cannot be converted to SQL

tjbanghart committed:
[CALCITE-5449] Allow EXTRACT() to accept DAYOFWEEK, DAYOFYEAR, ISOWEEK, WEEK(WEEKDAY) (#32)

* [CALCITE-5449] Allow EXTRACT() to accept time frames

* Fix getMonotonicity() override

* Lint

* Refactor DAYOFWEEK, DAYOFYEAR

---------

Co-authored-by: tanclary <116591231+tanclary@users.noreply.github.com>
Co-authored-by: Tanner Clary <tannerclary@google.com>

wnob committed:
Fix for sorting by literals

tanclary committed:
[CALCITE-6001] Add dialect-specific encoding for string literals

olivrlee committed:
Fixup for MetaColumn constructor changes in [CALCITE-5549]

Fix LintTest and RelWriterTest after fixup

wnob committed:
[CALCITE-6055] Customize handling of type name based on type system

wnob committed:
[CALCITE-6062] Parse timestamps more permissively

julianhyde commented:
Since literals also exist in source SQL, and intermediate SqlNode and RexNodes, you should change the summary to make clear that this relates to generating SQL. 'JDBC adapter should...' is one way.

wnob committed:
[CALCITE-5508] Allow no-op versions of DATETIME and TIMESTAMP

tanclary committed:
[CALCITE-6196] Accept BigQuery PERCENTILE functions without OVER clause

Address Mihai comment

tanclary committed:
Revert "[CALCITE-6006] RelToSqlConverter loses charset information"

This reverts commit 485a5d0ae3274a83a36788290b3bdc4d973387b4.

mihaibudiu commented on core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:
```diff
@@ -1726,6 +1726,9 @@ private static String toSql(RelNode root, SqlDialect dialect,
     assertThat(toSql(root), isLinux(expectedSql));
   }
 
+
```
maybe you want to take these empty lines back

mihaibudiu commented on core/src/main/java/org/apache/calcite/sql/SqlDialect.java:
```diff
@@ -1664,7 +1700,20 @@ private ContextImpl(DatabaseProduct databaseProduct,
+      return charset;
+    }
+
+    @Override public Context withCharset(String charset) {
```
does this qualify as a breaking change which has to be documented in history.md?

julianhyde replied:
I don't think so. ContextImpl is private.

zabetak committed:
[CALCITE-6608] RexBuilder#makeIn should create EQUALS instead of SEARCH for single point values

Close apache/calcite#3991

tisonkun committed:
Chore: Fix STDDEV inline comments

NobiGo committed:
[CALCITE-6631] The common type for a comparison operator returns the wrong type when comparing a Java type long with a SQL type INTEGER

sreeharshar84 committed:
[CALCITE-6633] Add CEILING to MSSQL dialect

Indentation and minor changes.

[CALCITE-6633] Add CEILING to MSSQL dialect

Add Jira link to test.

[CALCITE-6633] Add CEILING to MSSQL dialect

Add CEILING to MSSQL dialect

NobiGo committed:
[CALCITE-6638] Optimization that simplifies expressions such as '1 > a or 1 <= a or a is null' to TRUE is incorrect when it requires casts that are not lossless

NobiGo committed:
[CALCITE-5156] Support implicit integer types cast for IN Sub-query
